### PR TITLE
choicesLimit for RichChoiceView

### PR DESCRIPTION
**Pull request by microArtur**
https://nanopay.atlassian.net/browse/NP-8754

### CODE REVIEW

File: src/foam/u2/view/RichChoiceView.js [hunk 1]
```diff
@@ -66,6 +66,15 @@ foam.CLASS({
...
       name: 'heading',
       documentation: 'The heading text for this section.'
+    },
+    {
+      class: 'Int',
+      name: 'choicesLimit'
+    },
+    {
+      class: 'Boolean',
+      name: 'refineInput_',
```
Comment (kgrgreer) [inline src/foam/u2/view/RichChoiceView.js:76]: Can you document what this does?

File: src/foam/u2/view/RichChoiceView.js [hunk 2]
```diff
@@ -583,6 +599,10 @@ foam.CLASS({
                                   })
                                 .end();
                             }, false, self.comparator)
+                          .end()
+                          .start()
+                            .add(section.refineInput_$.map(v => v ? self.MORE_CHOICES : ''))
+                            .style({padding: '8px 16px'})
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/RichChoiceView.js]: Should be using a cssClass instead of style

File: src/foam/u2/view/RichChoiceView.js [hunk 3]
```diff
@@ -583,7 +605,13 @@ foam.CLASS({
                                   })
                                 .end();
                             }, false, self.comparator)
-                          .end();
+                          .end()
+                          .callIf(section.choicesLimit, function() {
```
Comment (microArtur) [inline src/foam/u2/view/RichChoiceView.js:609]: I hope this is the right way

File: src/foam/u2/view/RichChoiceView.js [hunk 4]
```diff
@@ -583,7 +605,13 @@ foam.CLASS({
...
+                            this.start()
+                              .addClass('moreChoices')
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/RichChoiceView.js]:
```suggestion
                              .addClass(this.myClass('moreChoices'))
```

File: src/foam/u2/view/RichChoiceView.js [hunk 5]
```diff
@@ -270,6 +284,10 @@ foam.CLASS({
       color: $destructive400;
       cursor: pointer;
     }
+
+    ^ .moreChoices {
```
Comment (sarthak-marwaha) [inline src/foam/u2/view/RichChoiceView.js]:
```suggestion
    ^moreChoices {
```

**Review by sarthak-marwaha — APPROVED** [review]

**Pull request merged**